xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h

```diff
@@ -2029,7 +2055,7 @@ class ATTR_DLL_LOCAL CFile
     // past the number of bytes read, we read (and skipped) a newline sequence.
     char buffer[1025];
     if (CPrivateBase::m_interface->toKodi->kodi_filesystem->read_file_string(
-            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer)))
+            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer) - 1))
```

Comment:
Before assigning buffer to line, you should do `buffer[1024] = '\0';`

Reply:
Nice catch, or 1025 even.

Reply:
Check my message about the strncpy idiom that I just posted. According to the idiom, this should be 1024 (1025 is a buffer overflow).

Reply:
Fixed

Reply:
Yes, correct.

Reply:
Really fixed now ;)

Reply:
Cool, now we want to make sure the comment above this line is correct. The time has come to _gasp_ fulfill a TODO.

So we:

* Drop TODO
* _Read 1024 chars into buffer_ is correct
* _If file position advanced that many chars, we didn't hit a newline_ is also correct I think
*  _Otherwise, if file position is 1 or 2 past the number of bytes read, we read (and skipped) a newline sequence_ Can file position go 1 or 2 past the number of bytes read?
* Add a note that the strncpy idiom is used here, and maybe even paste the idiom verbatim to save new readers a trip to stack overflow and educate them on how C-style reads should work elsewhere in the codebase.

```c
// char buffer[BUFFER_SIZE];
// strncpy(buffer, sourceString, BUFFER_SIZE - 1);
// buffer[BUFFER_SIZE - 1] = '\0';
```